### PR TITLE
Add initial documentation on Unit Testing and UI Testing

### PR DESCRIPTION
**Pull request by michael-hawker**
Adds info all about our testing setup from unit tests to UI tests!

@RosarioPulella would love any insights you have on this side of things. Feeling like I've probably missed something, but tried to cover a range of everything. Let me know your thoughts!

### CODE REVIEW

File: Testing.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+To setup your UI element, you can construct it manually or call `XamlReader.Load` with an XML string to represent your test setup.
+
+### Good Testing Practices
+
+It is important to test a variety of scenarios when implementing a component. These should cover not just success scenarios, but failure scenarios and edge cases as well. Try and think about how your component shouldn't be used and right tests to ensure guards against that work as intended or exceptions are thrown.
+
+Use `[ExpectedException]` and `[DataRow]` attributes to help you easily test exceptions or a variety of scenarios with the same test code.
+
+When fixing a bug, it can also be handy to write a unit test first which shows the broken scenario, and then validate the bug is fixed by writing the solution against the same test (which should now pass).
+
```
Comment (RosarioPulella) [inline Testing.md:52]: Maybe link to https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices here?

**Review by RosarioPulella — APPROVED** [review]
Overall it looks really good. I think we provide enough info about the `UITests` so that most contributors could read this and figure it out.

File: Testing.md (hunk 1 above)
Comment (michael-hawker) [reply Testing.md:52]: Awesome! Done, thanks!
